app/screens/node/Node.tsx

```diff
@@ -360,12 +355,8 @@ const Node = ({ history, location }: Props) => {
     (state: RootState) => state.smesher.maxFileSize
   );
 
-  const rewards = useSelector((state: RootState) =>
-    state.smesher.rewards.slice(0).reverse()
-  );
-  const rewardsInfo = useSelector(
-    (state: RootState) => state.smesher.rewardsInfo
-  );
+  const rewards = useSelector(SmesherSelectors.getRewards);
+  const rewardsInfo = useSelector(SmesherSelectors.getRewardsInfo);
```

Comment:
For this selector, because of calculations, we can consider using memoization. It might be better to separate the functions, cover them with tests, and then have the selectors return only raw data. Alternatively, you can write your own hook for reuse or use useMemo here to avoid unnecessary recalculations.

Reply:
Yeah, it might help a bit with re-opening Smesher screen (if memoization will be done in a global scope, not in the component). But it won't help with the case when Node is actively syncing (a lot of new rewards are coming every second).
So I will add memoization here and then try to optimize updating the list of rewards (and avoid re-calculating from scratch every time)...